tests/telemetry/events/host_events.py

```diff
@@ -0,0 +1,36 @@
+#! /usr/bin/env python3
+
+import logging
+
+from run_events_test import run_test
+from event_utils import backup_monit_config, customize_monit_config, restore_monit_config
+
+logger = logging.getLogger(__name__)
+tag = "sonic-events-host"
+
+
+def test_event(duthost, gnxi_path, ptfhost, data_dir, validate_yang):
+    logger.info("Beginning to test host events")
+    backup_monit_config(duthost)
+    customize_monit_config(
+        duthost,
+        [
+            "> 90% for 10 times within 20 cycles then alert repeat every 1 cycles",
+            "> 2% for 1 times within 5 cycles then alert repeat every 1 cycles"
+        ]
+    )
+
+    run_test(duthost, gnxi_path, ptfhost, data_dir, validate_yang, None,
+             "memory_usage.json", "sonic-events-host:memory-usage", tag, False)
+    run_test(duthost, gnxi_path, ptfhost, data_dir, validate_yang, None,
+             "disk_usage.json", "sonic-events-host:disk-usage", tag, False)
+    run_test(duthost, gnxi_path, ptfhost, data_dir, validate_yang, None,
+             "cpu_usage.json", "sonic-events-host:cpu-usage", tag, False)
+    run_test(duthost, gnxi_path, ptfhost, data_dir, validate_yang, trigger_mem_threshold_exceeded_alert,
+             "mem_threshold_exceeded.json", "sonic-events-host:mem-threshold-exceeded", tag)
+    restore_monit_config(duthost)
```

Comment:
>restore_monit_config [](http://example.com/codeflow?start=4&length=20)

If anything above fails, will you restore?

Reply:
I will add.

Reply:
Done

Reply:
I see you add new callback into run_test function. Actually, you may implement by a big `try-finally` in this function. No need to add it recursively.

Reply:
Makes sense, will change

Reply:
Done